node/service/src/lib.rs

```diff
@@ -674,23 +708,50 @@ where
 	let disable_grandpa = config.disable_grandpa;
 	let name = config.network.node_name.clone();
 
-	let service::PartialComponents {
+	let overseer_connector = OverseerConnector::default();
+
+	let handle = Handle(overseer_connector.as_handle().clone());
+
+	let basics = new_partial_basics::<RuntimeApi, ExecutorDispatch>(
+		&mut config,
+		jaeger_agent,
+		telemetry_worker_handle,
+	)?;
+
+	// we should remove this check before we deploy parachains on polkadot
+	// TODO: https://github.com/paritytech/polkadot/issues/3326
+	let chain_spec = &config.chain_spec as &dyn IdentifyVariant;
+
+	let is_relay_chain = chain_spec.is_kusama() ||
+		chain_spec.is_westend() ||
+		chain_spec.is_rococo() ||
+		chain_spec.is_wococo();
```

Comment:
Looks very easy to forget adding a chain - what would be the consequences?

Reply:
We would use the `LongestChainSelection`-scheme instead of the one taking disputes into account. We currently don't have an alternative way of extracting this (well, we could add another trait `IsRelayChain` but that would just defer the problem elsewhere)

Reply:
This is a temporary thing that goes away once we have parachains on polkadot, see the issue linked above.
It wasn't introduced in this code, just moved.

Reply:
Consequences of forgetting would be pretty bad though. We could maybe make parachains not work, if `is_relay_chain` is false?

Reply:
Consequences are that we'd finalized blocks as if there were no parachains. You can think of this an optimization of finality lag xD